mne/viz/epochs.py

```diff
@@ -536,3 +724,172 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, proj=False, n_fft=256,
     if show:
         plt.show()
     return fig
+
+
+def _plot_traces(params):
+    """ Helper for plotting concatenated epochs """
+    n_channels = params['n_channels']
+    types = params['types']
+    lines = params['lines']
+    data = params['data']
+    offsets = params['offsets']
+
+    tick_list = []
+    start_idx = int(params['t_start'] / len(params['epochs'].times))
+    labels = params['labels'][start_idx:]
+    params['ax'].set_xticklabels(labels)
+    # do the plotting
+    for ii in range(n_channels):
+        ch_ind = ii + params['ch_start']
+        if ii >= len(lines):
+            break
+        elif ch_ind < len(params['picks']):
+            # scale to fit
+            ch_name = params['ch_names'][ch_ind]
+            tick_list += [ch_name]
+            offset = offsets[ii]
+            end = params['t_start'] + params['duration']
+            this_data = data[ch_ind][params['t_start']:end]
+
+            # subtraction here gets correct orientation for flipped ylim
+            ydata = offset - this_data
+            xdata = params['times'][:params['duration']]
+            segments = np.split(np.array(zip(xdata, ydata)),
+                                params['n_epochs'])
+
+            lines[ii].set_segments(segments)
+            vars(lines[ii])['ch_name'] = ch_name
+            color = params['color'][types[ch_ind]]
+            vars(lines[ii])['def_color'] = params['color'][types[ch_ind]]
+            lines[ii].set_color(color)
+        else:
+            # "remove" lines
+            lines[ii].set_xdata([])
```

Comment:
Something is buggy here @jaeilepp . I get this error after using the keyboard shortcuts for a while:

```py
Traceback (most recent call last):
  File "/home/mainak/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 315, in keyPressEvent
    FigureCanvasBase.key_press_event(self, key)
  File "/home/mainak/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1816, in key_press_event
    self.callbacks.process(s, event)
  File "/home/mainak/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 540, in process
    proxy(*args, **kwargs)
  File "/home/mainak/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 415, in __call__
    return mtd(*args, **kwargs)
  File "mne/viz/epochs.py", line 854, in _plot_onkey
    _plot_traces(params)
  File "mne/viz/epochs.py", line 767, in _plot_traces
    lines[ii].set_xdata([])
AttributeError: 'LineCollection' object has no attribute 'set_xdata'

```